sway-lib-std/src/bytes.sw

```diff
@@ -871,6 +872,86 @@ impl Bytes {
 
         self.len = new_len;
     }
+
+    /// Returns an [Iterator] to iterate over this `Bytes`.
+    ///
+    /// # Returns
+    ///
+    /// * [BytesIter<V>] - The struct which can be iterated over.
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let mut bytes = Bytes::new();
+    ///     bytes.push(5);
+    ///     bytes.push(10);
+    ///     bytes.push(15);
+    ///
+    ///     // Get the iterator
+    ///     let iter = bytes.iter();
+    ///
+    ///     assert_eq(5, iter.next().unwrap());
+    ///     assert_eq(10, iter.next().unwrap());
+    ///     assert_eq(15, iter.next().unwrap());
+    ///
+    ///     for elem in bytes.iter() {
+    ///         log(elem);
+    ///     }
+    /// }
+    ///
+    /// # Undefined Behavior
+    ///
+    /// Modifying vector during iteration is a logical error and
+    /// results in undefined behavior. E.g.:
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let mut bytes = Bytes::new();
+    ///     bytes.push(5);
+    ///     bytes.push(10);
+    ///     bytes.push(15);
+    ///
+    ///     for elem in bytes.iter() {
+    ///         bytes.push(20); // Modification causes undefined behavior.
+    ///     }
+    /// }
+    /// ```
+    pub fn iter(self) -> BytesIter {
+        // WARNING: Be aware of caveats of this implementation
+        //          if you take it as an example for implementing
+        //          `Iterator` for other types.
+        //
+        //          Due to the Sway's copy semantics, the `values` will
+        //          actually contain **a copy of the original vector
```

Comment:
Sentences shouldn't start on a newline so forc doc can handle it correctly

Reply:
These are internal comments not docs. I took these from the vector implementation